fluentd/generate_syslog_config.rb

```diff
@@ -0,0 +1,54 @@
+# This ruby script creates the dynamic configuration file 
+# needed for the fluentd-remote-syslog plugin.
+
+@env_vars = Array.new
+
+# Loops through environment variables for the remote-syslog plugin
+def init_environment_vars()
+  prefix = 'REMOTE_SYSLOG_'
+  group  = prefix + "HOST"
+
+  ENV.each_key.to_a.select { |s| s.start_with? group }.uniq.each do |k|
+    # [0] = environment variable name
+    # [1] = syslog plugin config name
+    # [2] = default value, if any
+    vars = [
```

Comment:
Seems this should only need to be done once and definitely more readable if it was a hash or collection of hashes
```
vars: {
  'HOST' : nil,
 'PORT' : 514
....
}
```

Reply:
It's a little complicated.  It's an array of Hashes because it can handle the case if the user passes in multiple ENV variables to configure mulitple syslog servers.  Any ENV variables that start with the prefix "REMOTE_SYSLOG" will be grouped based on their suffix.

For example, you could set REMOTE_SYSLOG_HOST=192.168.1.1 and REMOTE_SYSLOG_HOST_2=192.168.1.2 as ENV variables.  This script will build two fluentd \<store\> configurations.  Group suffixes are any characters after the string "REMOTE_SYSLOG_HOST" - as that is the single required field to build any configuration at all.

So above, the two groups would be "" and "_2".  That's why it's an array of Hashes.